lock-api/src/main/java/com/palantir/lock/DebugThreadInfoConfiguration.java

```diff
@@ -16,16 +16,23 @@
 
 package com.palantir.lock;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import org.immutables.value.Value;
 
+@JsonDeserialize(as = ImmutableDebugThreadInfoConfiguration.class)
+@JsonSerialize(as = ImmutableDebugThreadInfoConfiguration.class)
 @Value.Immutable
 public interface DebugThreadInfoConfiguration {
 
+    @JsonProperty("record-thread-info")
```

Comment:
Q for reviewer: I copied this style from TimeLockRuntimeConfiguration. Is this appropriate?

Reply:
That's fine, the reason for this is b/c otherwise we use camelcase, but internally we use kebab case everywhere